### DISCUSSION TITLE
Token expiry

### DISCUSSION THREAD

**Discussion by cdpeac** [isFakeAccount/psnawp#58] [Q&A] [answered]
I note in the docs that the bot will warn you when your token is going to expire.  Is there some way we can monitor that programmatically?

**Comment by isFakeAccount** [comment, marked as answer]
In version 2.0.0 I made it easier to get the refresh token expiration time. See the documentation 

https://psnawp.readthedocs.io/en/latest/psnawp_api.core.html#psnawp_api.core.authenticator.Authenticator.refresh_token_expiration_time

```py
psn = PSNAWP(getenv("NPSSO_CODE", "NPSSO_CODE"))
print(psn.authenticator.refresh_token_expiration_time)
```
This should work and you can write code to notify you at this time.
